src/lambda/expression.rs

```diff
@@ -79,6 +79,22 @@ impl Expression {
               let is_get = req.method() == reqwest::Method::GET;
               // Attempt to short circuit GET request
               if is_get {
+                if let Some(data_loader) = ctx.req_ctx.data_loader.as_ref() {
```

Comment:
Re-write the logic should be as follows — 
- Check if it's a BATCH Compatible request (method is GET, and parent was a list)
- if TRUE: Use data-loader to make HTTP calls
- if FALSE: Use http-client to make HTTP calls

Drop all unnecessary caching logic from here.